src/test/kotlin/no/nav/syfo/client/arkivporten/ArkivportenClientTest.kt

```diff
@@ -0,0 +1,69 @@
+package no.nav.syfo.client.arkivporten
+
+import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
+import io.mockk.mockk
+import java.util.*
+import kotlin.test.assertEquals
+import kotlin.test.assertTrue
+import kotlinx.coroutines.runBlocking
+import no.nav.syfo.infrastructure.client.arkivporten.ArkivportenClient
+import no.nav.syfo.infrastructure.client.arkivporten.ArkivportenClientException
+import no.nav.syfo.infrastructure.client.arkivporten.ArkivportenDocument
+import no.nav.syfo.infrastructure.client.azuread.AzureAdV2Client
+import no.nav.syfo.testhelper.ExternalMockEnvironment
+import no.nav.syfo.testhelper.UserConstants
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertNotNull
+
+class ArkivportenClientTest {
+    fun arkivportenDocument(orgNumber: String) = ArkivportenDocument(
+        documentId = UUID.randomUUID(),
+        type = ArkivportenDocument.DocumentType.DIALOGMOTE,
+        content = byteArrayOf(23, 45, 67, 89),
+        contentType = ArkivportenDocument.ContentType.PDF,
+        orgNumber = orgNumber,
+        title = "Test dialogmøte",
+        summary = "Dialogmøte opprettet den 01.01.2024",
+        fnr = UserConstants.ARBEIDSTAKER_FNR.value,
+        fullName = "Test Person"
+    )
+
+    val azureAdV2ClientMock = mockk<AzureAdV2Client>(relaxed = true)
+    val externalMockEnvironment = ExternalMockEnvironment.getInstance()
+    val arkivportenClient = ArkivportenClient(
+        azureAdV2Client = azureAdV2ClientMock,
+        baseUrl = externalMockEnvironment.environment.arkivportenUrl,
+        clientId = externalMockEnvironment.environment.arkivportenClientId,
+        client = externalMockEnvironment.mockHttpClient,
+    )
+
+    @Test
+    fun `sends document successfully`() {
+        runBlocking {
+            arkivportenClient.sendDocument(arkivportenDocument(UserConstants.VIRKSOMHETSNUMMER_HAS_NARMESTELEDER.value))
+        }
+    }
+
+    @Test
+    fun `throws ArkivportenClientException when API returns error`() {
+        var exception: ArkivportenClientException? = null
+        runBlocking {
+            try {
+                arkivportenClient.sendDocument(arkivportenDocument(UserConstants.VIRKSOMHETSNUMMER_ARKIVPORTEN_FAILS.value))
+            } catch (e: ArkivportenClientException) {
+                exception = e
+            }
```

Comment:
Kan kan også bruke `assertThrows` eller `assertFailsWith` for lignende test. Eksempel her: https://github.com/navikt/ismeroppfolging/blob/ee0fd621cf88d01388c7b4ff435e477ed613b5e8/src/test/kotlin/no/nav/syfo/senoppfolging/domain/SenOppfolgingKandidatTest.kt#L31